### PR TITLE
Image Analysis SDK: Add JavaDoc code snippets. Update CODEOWNERS.

### PR DESCRIPTION
**Pull request by dargilco**
* Add code snippets in JavaDocs for the ImageAnalysisClientBuilder and the analyze methods of ImageAnalysisClient.
* Add CI pipeline variable EnableVerifyTypeSpecCodeGeneration that allows me to disable the pipeline task "Verify Type Spec Code Generation" for this SDK. The verification tool needs to be update to take into account code-snippets inserted into Java source code comments (JavaDocs). Until this is done, I need to disable this task. But I did verify before disabling this task, that the only diff it found was related to code-snippets in Java source code comments.
* Rename method ImageAnalysisOptions.getGenderNeutralCaption() to isGenderNeutralCaption(), per guidelines for Boolean properties
* Fix  ServiceLabel in CODEOWNERS
* Remove pom.xml and scripts from the samples folder

### CODE REVIEW

**Review by rhurey — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

File: sdk/vision/ci.yml
Comment (srnagar) [inline sdk/vision/ci.yml:45]: File a GitHub issue to track this work item and link this to the backlog issue in AutoRest Java repo. So, when the codegen changes are ready, we will know that this task is unblocked. 

Also, another thing we can do (in a separate PR) is to not skip this step entirely but to not fail the build on error instead (by using [ContinueOnError](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-task?view=azure-pipelines)). So, it won't block merging PR but it will also remind us that there are TypeSpec changes that don't match which can prompt devs to look and validate if it's just code snippets or if there's more.

Comment (weidongxu-microsoft) [reply sdk/vision/ci.yml:45]: Codegen issue on https://github.com/Azure/autorest.java/issues/2492

File: sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/SampleOcrImageUrlAsync.java
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/SampleOcrImageUrlAsync.java]: Thanks @srnagar, I made the changes based on the examples you provided.

File: sdk/vision/ci.yml
Comment (dargilco) [reply sdk/vision/ci.yml]: Good catch @srnagar. Fixed!

File: sdk/vision/ci.yml
Comment (dargilco) [reply sdk/vision/ci.yml]: Opened https://github.com/Azure/azure-sdk-for-java/issues/38337 . Thanks!

File: .github/CODEOWNERS
Comment (dargilco) [reply .github/CODEOWNERS:477]: @weshaggard none of the ServiceLabel tags have a folder associated with them, so I removed what I had there before, thinking it should be like all other ServcieLabel. We do have a PRLabel with our folder:
```
# PRLabel: %Image Analysis
/sdk/vision/azure-ai-vision-imageanalysis/           @dargilco @rhurey
```
Perhaps I misunderstood what ServiceLabel tag means and what the folder there means, and how it is different than PRLabel.

Comment (weshaggard) [reply .github/CODEOWNERS:477]: Oh you have that in a different spot. You can combine them and add the ServiceLabel in your other section. See https://github.com/Azure/azure-sdk-tools/blob/main/tools/codeowners-utils/METADATA.md for some examples.

Comment (dargilco) [reply .github/CODEOWNERS:477]: Thanks Wes! I saved the link and added a note to look into this as part of my next PR.

**Pull request merged**
